internal/handlers/common/params.go

```diff
@@ -26,8 +26,11 @@ import (
 	"github.com/FerretDB/FerretDB/internal/util/must"
 )
 
-// GetRequiredParam returns doc's value for key
-// or protocol error for missing key or invalid type.
+// GetRequiredParam returns doc's value for key.
+//
+// Returned errors:
+//   - ErrBadValue if the document value is missing;
+//   - ErrBadValue if the document value is not of the expected type.
 func GetRequiredParam[T types.Type](doc *types.Document, key string) (T, error) {
 	var zero T
```

Comment:
`defer contract.EnsureError(err, ErrBadValue)`

Reply:
It seems that I have to specify that it's not "errors" but error types. We can check that those errors have `CommandError` type but there is no way to check the error code now. We could possibly extend the `contract` package with `EnsureCommandError` and `EnsureWriteError`. What do you think?

Reply:
That's what I tried to do – and that's surprisingly not easy.

Let's only update documentation for now. `ErrBadValue` is not an error and also not a error type

Reply:
I like this. I think this is the practice we can expand for new code bases :hugs: